docs/contributing/999-rfcs/2024-01-29-tests.md

```diff
@@ -0,0 +1,300 @@
+---
```

Comment:
This is an initial version, and I haven't filled some of the fields, @winglang/maintainers feel free to commit/ comment suggestions and solutions